### ISSUE TITLE
[dev.to] Titles in comments end up in the TOC

### ISSUE THREAD

**Issue by bcouetil** [derlin/bitdowntoc#16]
Thank you again for this great tool, I'm using on every dev.to articles 🥰

Today I came accross a bug : titles in comments end up in the TOC

By comments I mean `<!-- -->`

**Comment by derlin** [comment]
Thank you for the kind words! Happy to hear you use it. 
Can you provide a minimal reproducible example? So I can better understand the problem.

**Comment by derlin** [comment]
Actually, I believe HTML comments are not supported by the  devto parser, you need to use liquid tags to add comments. Could you try the comment elsewhere in the article and see if it is rendered as well?

Finally, can you confirm you use the devto profile? Html comments are only used in others.

**Comment by bcouetil** [comment]
Sure, happy to help. No mistake, this is `dev.to` that I choose for generation.

Here is the code, including the generation :

```markdown
---
title: "Comments Test"
license: public-domain
published: false
---

{%- # TOC start (generated with https://github.com/derlin/bitdowntoc) -%}

- [Introduction](#introduction)
- [Conclusion](#conclusion)
- [Hello world](#hello-world)
- [Ohayou sekai](#ohayou-sekai)

{%- # TOC end -%}

# Introduction

Lorem ipsum.

<!--

# Hello world

# Ohayou sekai

-->

# Conclusion

Dolor sit amet.
```

And here is the [preview](https://dev.to/bcouetil/comments-test-24je-temp-slug-536761?preview=bc571ceb7477081ea405eeb52bc97bc796e3256d5feb8ee36926dc7c3cf32c0e1682e04d3505d453f52ab644731d82ccc31db712a25489c68142bcdc).

Visual Studio Code makes makes HTML comments when I ask it to comment out something.

And I did not find anywhere what is the proper way of commenting on dev.to.

I carefully read the [documentation about liquid tags](https://developers.forem.com/frontend/liquid-tags) and did not find anything about commenting out code. The `devcomment` is for embedding actual comments 🧐

**Comment by derlin** [comment]
Well, just look at the comments I use in the TOC then 😉. To start and to end don't render do they? 

```
{%- this is a liquid comment -%}
```

**Comment by bcouetil** [comment]
You are right, they are so in front of my eyes that I forget their existence 😅

I'll use that from now on.

Where did you find that it is better than HTML comments ?

**Comment by derlin** [comment]
Assuming by better you mean how did I find out how to comment on dev.to, I googled "liquid tag comments" and stumbled very quickly upon the `{% %}` syntax, e.g. here https://shopify.dev/docs/api/liquid/tags.

The problem though is that it leaves newlines. I looked up "whitespace control liquid" and got the rule:

> If you don’t want any of your tags to print whitespace, as a general rule you can add hyphens to both sides of all your tags (`{%- and -%}`)

Combining the two made it work as expected.
